### ISSUE TITLE
Performance of inotify_simple

### ISSUE THREAD

**Issue by SteelBlueVision** [chrisjbillington/inotify_simple#8]
There are several pythonic (and perhaps non-pythonic) ways that the performance of inotify_simple could be increased without breaking its "philosophy." It is hard to say by how much, without actually trying. Would you be open to such changes via a pull request or do I need to create a whole new package based on yours?

**Comment by chrisjbillington** [comment]
Yes, I am definitely open to accepting pull requests for performance enhancements, pythonic or otherwise. I am a bit sceptical that worthwhile gains can be made without things like Cython (and I love Cython, but not requiring a compilation step is part of the "philosophy" of this package at present), but if you can find gains anywhere, that would be great - performance is part of the aims of the project too :)
